### ISSUE TITLE
loss is nan

### ISSUE THREAD

**Issue by passionvery** [HobbitLong/SupContrast#50]
the code:
mask.sum(1) has the number 0
so the loss is nan
Is there any problem with my code?
respect for your answer,thank you

**Comment by HobbitLong** [comment]
`mask.sum(1)` should be at least 1, if you are feeding features of two views of the same image following [here](https://github.com/HobbitLong/SupContrast/blob/master/losses.py#L27).

**Comment by sidml** [comment]
@passionvery 
You may want to check if your are using `F.normalize` before calculating the loss function [(like here](https://github.com/HobbitLong/SupContrast/blob/6d5a3de39070249a19c62a345eea4acb5f26c0bc/networks/resnet_big.py#L185)).

I missed this step and got nan.

**Comment by hmt2014** [comment]
@sidml 
It is a great answer. 
I have some confusions:
What is the effect of F.normalize ? It normalizes the feature dimension. How does it affects the loss ?

**Comment by sunxiaojie99** [comment]
> @sidml It is a great answer. I have some confusions: What is the effect of F.normalize ? It normalizes the feature dimension. How does it affects the loss ?

hi, I want to ask if you know the reason for using l2 regularization. When I don't use l2, the model will even explode in gradient 😭
